crypto/x509v3/v3_admis.c

```diff
@@ -0,0 +1,187 @@
+            if (BIO_printf(bp, "%*s  admissionAuthority:\n", ind, "") <= 0) goto err;
+            if (BIO_printf(bp, "%*s    ", ind, "") <= 0) goto err;
+            if (GENERAL_NAME_print(bp, entry->admissionAuthority) <= 0) goto err;
+            if (BIO_printf(bp, "\n") <= 0) goto err;
+        }
+
+        if (entry->namingAuthority != NULL) {
+            if (i2r_NAMING_AUTHORITY(method, entry->namingAuthority, bp, ind) <= 0) goto err;
+        }
+
+        for (j = 0; j < OPENSSL_sk_num((const OPENSSL_STACK*)entry->professionInfos); j++) {
+            PROFESSION_INFO* pinfo = sk_PROFESSION_INFO_value(entry->professionInfos, j);
+
+            if (BIO_printf(bp, "%*s  Profession Info Entry %0d:\n", ind, "", 1 + j) <= 0) goto err;
+
+            if (pinfo->registrationNumber != NULL) {
+                if (BIO_printf(bp, "%*s    registrationNumber: ", ind, "") <= 0) goto err;
+                if (ASN1_STRING_print(bp, pinfo->registrationNumber) <= 0) goto err;
+                if (BIO_printf(bp, "\n") <= 0) goto err;
+            }
+
+            if (pinfo->namingAuthority != NULL) {
+                if (i2r_NAMING_AUTHORITY(method, entry->namingAuthority, bp, ind + 2) <= 0) goto err;
+            }
+
+            if (pinfo->professionItems != NULL) {
+                ASN1_STRING* val;
+
+                if (BIO_printf(bp, "%*s    Info Entries:\n", ind, "") <= 0) goto err;
+                for (k = 0; k < sk_ASN1_STRING_num(pinfo->professionItems); k++) {
+                    if (BIO_printf(bp, "%*s      ", ind, "") <= 0) goto err;
+
+                    val = sk_ASN1_STRING_value(pinfo->professionItems, k);
+                    if (ASN1_STRING_print(bp, val) <= 0) goto err;
+                    if (BIO_printf(bp, "\n") <= 0) goto err;
+                }
+            }
+
+            if (pinfo->professionOIDs != NULL) {
+                if (BIO_printf(bp, "%*s    Profession OIDs:\n", ind, "") <= 0) goto err;
+                for (k = 0; k < sk_ASN1_OBJECT_num(pinfo->professionOIDs); k++) {
+                    ASN1_OBJECT* o = sk_ASN1_OBJECT_value(pinfo->professionOIDs, k);
+                    const char *ln = OBJ_nid2ln(OBJ_obj2nid(o));
+                    char objbuf[128];
+
+                    OBJ_obj2txt(objbuf, sizeof(objbuf), o, 1);
+                    if (BIO_printf(bp, "%*s      %s%s%s%s\n", ind, "", 
+                                   ln ? ln : "", ln ? " (" : "",
+                                   objbuf, ln ? ")" : "") <= 0) 
+                        goto err;
+                }
+            }
+        }
+    }
+
+    return 1;
+err:
+    return -1;
+}
+
+static X509V3_EXT_METHOD v3_ext_admission = {
+    /* .ext_nid = */ NID_x509ExtAdmission,
+    /* .ext_flags = */ 0,
+    /*.it = */  ASN1_ITEM_ref(ADMISSION_SYNTAX),
+    NULL, NULL, NULL, NULL,
+    /*.i2s = */ NULL,
+    /*.s2i = */ NULL,
+    /*.i2v = */ NULL,
+    /*.v2i = */ NULL,
+    /*.i2r = */ &i2r_ADMISSION_SYNTAX,
+    /*.r2i = */ NULL,
```

Comment:
what's that crazy comment syntax? :)  put the field name *after* the field value, or leave it off completely.
leave off trailing NULL elements.  and leave off trailing comma.
look at ocsp/v3_ocsp.c for formatting example.